### PR TITLE
test: add end-to-end smoketest for poetry

### LINKED ISSUE #199: appmap-python has an end-to-end test that uses poetry

### PR DESCRIPTION
**Pull request by apotterri**
Add a quick and dirty end-to-end smoketest that uses poetry.

Fixes #199 .

### CODE REVIEW

File: ci/test_poetry.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+set -e
+pip install poetry
+
+mkdir /poetry || true
+cd /poetry
+
+poetry init -q
+
+# Yes, we need to set RUNNER, and we need to "poetry run" the script.
```
Comment (dividedmind) [inline ci/test_poetry.sh:11]: Good call adding this comment :)

**Review by dividedmind — CHANGES_REQUESTED** [review]
It's a start, but perhaps a bit too quick and dirty — I'd like to see the test using `poetry add`, for example; as it is it basically only tests the poetry virtualenv, but not its package dependency machinery (if I understand it correctly). Unless you don't think it's worth it?

**Comment by apotterri** [comment]
I'm not sure it is worth it. With this implementation, it at least shows that installing appmap doesn't break a poetry project, in the same way that test_pipenv.sh shows that it doesn't break a pipenv project.

This test would have prevented us from publising v1.12.4, a release which rendered a poetry project completely non-functional.

**Comment by dividedmind** [comment]
> I'm not sure it is worth it. With this implementation, it at least shows that installing appmap doesn't break a poetry project, in the same way that test_pipenv.sh shows that it doesn't break a pipenv project.
> 
> This test would have prevented us from publising v1.12.4, a release which rendered a poetry project completely non-functional.

Ok, this makes sense. Thanks for explaining the rationale, I'll approve!

**Review by dividedmind — APPROVED** [review]

**Pull request merged**